compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -155,7 +155,7 @@ void ConfigurationNotifier::CreateFactoryComponent(
   // component except the factory component itself.
   newMetadata->configurationPids.clear();
   for (const auto& basePid : oldMetadata->configurationPids) {
-    if (basePid != factoryName) {
+    if (basePid != oldMetadata->configurationPids[0])  {
```

Comment:
not sure I understand this change. can you elaborate?

Also, `factoryName` is no longer used which will cause a build failure on linux/mac

Reply:
The first pid in the configuration-pids for the factory component is the one used to create the factory instances. When creating the list of configuration-pids for the factory instance, it needs to be deleted from the list of configuration-pids and replaced with the specific configuration-pid that matches this factory instance. When I originally wrote this code I thought the pid used to create the factory instances was supposed to match the factory component name. That assumption was incorrect. Users have been using pids that do not match the factory component name to create the factory instances. So it's not correct to use the factoryName for the comparison here. I noticed this when testing the current change because I used a pid that was different than the factory component name to create the factory instances.